### PR TITLE
Add Core-side software versions for preview 1 of v21 release.

### PR DESCRIPTION
**Pull request by dmkozh**

### CODE REVIEW

File: docs/reference/software-versions.mdx [hunk 1]
```diff
@@ -14,6 +14,50 @@ Release candidates are software releases that are also released to the [Testnet]
 
 [testnet]: ./networks.mdx
 
+## Protocol 21: Preview 1 (Testnet only, April 12, 2024)
+
+### Software
+
+| Software | Version |
+| --- | --- |
+| XDR | `v21.0.1` |
```
Comment (anupsdf) [inline docs/reference/software-versions.mdx:23]: stellar-xdr also has a release https://github.com/stellar/stellar-xdr/releases/tag/v21.1. Should we add that and separate from rs-stellar-xdr?

**Review by anupsdf — APPROVED** [review]
LGTM

**Comment by anupsdf** [comment]
Preview page for this is here, https://developers-pr547.previews.kube001.services.stellar-ops.com/docs/reference/software-versions

**Review by rice2000 — APPROVED** [review]
Thanks!  LGTM!

**Pull request merged**